File: Leetcode/math/504_Base7/Solution.py
class Solution(object):
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        if not num: return '0'
        l, n = [], num < 0
        if n: num *= -1
        while num:
            l.append(str(num % 7))
            num //= 7
        if n: l.append('-')
        return ''.join(l[::-1])

File: Leetcode/math/504_Base7/Solution.java
class Solution {
    public String convertToBase7(int num) {
        if(num==0) return "0";
        StringBuilder res = new StringBuilder();
        char sign = '+';
        if(num<0){
            sign = '-';
            num *= -1;
        }
        while(num>0){
            res.append(num%7);
            num /= 7;
        }
        if(sign=='-') res.append(sign);
        return res.reverse().toString();
    }
}
